File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        i=0
        maxi=0
        j=len(nums)-1
        while(i<j):
            if maxi<min(nums[i],nums[j])*(j-i):
                maxi=min(nums[i],nums[j])*(j-i)
            if nums[i]<nums[j]:
                i+=1
            else:
                j-=1
        return maxi

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& hei) {
        int i = 0 , j = hei.size()-1 , ans = 0;
        while(i <= j){
            if(hei[i] < hei[j]){
                ans = max(ans , min(hei[i] , hei[j])*(j-i));
                i++;
            }
            else if(hei[i] >= hei[j]){
                ans = max(ans , (min(hei[i] , hei[j])*(j-i)));
                j--;
            }
            // else{
            //     ans = max(ans , min(hei[i] , hei[j]));
            //     i++;
            //     j--;
            // }
        }
        return ans;
    }
};